### PR TITLE
Support custom Provider

### PR DESCRIPTION
**Pull request by devrsi0n**
Add an option to support custom Provider.

### CODE REVIEW

**Comment by jescalan** [comment]
Hey @devrsi0n - this looks like a great addition, could you add a little more context on why this might be a helpful feature, and document it in the README as well? 🙏🏼

File: README.md [hunk 1]
```diff
@@ -47,8 +47,8 @@ While it may seem strange to see these two in the same file, this is one of the
...
+- `renderToString(source: string, components: object, options?: object)` - This function consumes a string of mdx along with any components it utilizes in the format `{ ComponentName: ActualComponent }`. The function returns an object that is intended to be passed into `hydrate` directly. It also can optionally be passed options which are [passed directly to mdx](https://mdxjs.com/advanced/plugins), `options.remarkPlugins` for reamrk plugins, `options.rehypePlugins` for rehype plugins, and `options.Provider` for custom provider (e.g. [emotion theme provider](https://emotion.sh/docs/theming))
+- `hydrate(source: object, components: object)` - This function consumes the output of `renderToString` as well as the same components argument as `renderToString`. Its result can be rendered directly into your component. This function will initially render static content, and hydrate it when the browser isn't busy with higher priority tasks.
```
Comment (devrsi0n) [inline README.md:51]: It seems `options` should place to `renderToString`, could you confirm that? @jescalan

Comment (jescalan) [reply README.md:51]: That's correct, this was fixed upstream though, so it can be rebased out

**Comment by jescalan** [comment]
This should be ready to go pending a rebase on upstream/master!

**Comment by ynnoj** [comment]
@jescalan @devrsi0n Will this enable me to use local React context inside m MDX components? I'm trying to use authentication/user state from context.

**Comment by jescalan** [comment]
@ynnoj - it would allow you to pass in a provider via the options, but would not automatically inherit from any provider above in the tree. another option is to pass in data using the `scope` argument to `renderToString`

**Comment by ynnoj** [comment]
> @ynnoj - it would allow you to pass in a provider via the options, but would not automatically inherit from any provider above in the tree. another option is to pass in data using the `scope` argument to `renderToString`

Local context wouldn't be available within the `getStaticProps` lifecycle though, right?

**Comment by jescalan** [comment]
@ynnoj That's correct - really the answer depends on how you're adding data to the context. I would be happy to work with you on figuring out how to solve this if you'd like, as I anticipate having the same problem myself down the line with auth/user context.

**Comment by ynnoj** [comment]
@jescalan Sure, that would be great! The application I'm working on is open source, so you an find the repo [here](https://github.com/firstmeanseverything/next).

I'm using Firebase Auth, so nothing crazy. But all client side, no SSR.

https://github.com/firstmeanseverything/next/blob/db3fa7637131742517c0c8abc52abfb6201a67c7/context/auth.js#L59-L63

**Comment by molebox** [comment]
Hey folks! Just stumbled across this PR and it would seem it might be an answer to a problem we are having with using mdx-embed with NEXT (via this package) Any movement on a merge?

**Comment by jescalan** [comment]
Yes, there will be a resolution to this issue shipped this week. Likely tonight.

**Comment by jescalan** [comment]
#35 supersedes this

**Pull request closed without merging**